### PR TITLE
Apply black formatting and add pre-commit hook

### LINKED ISSUE #5: Code Style
It would be a great idea to set up some automatic code-style checking to enable smoother collaboration. I am in favor of [black](https://black.readthedocs.io/en/stable/the_black_code_style.html) as it basically just makes it easy, especially when linked with [pre-commit](https://github.com/pre-commit/pre-commit).

Putting some `flake8` consistency checks into CI is also helpful for detecting code smells.

### PR DESCRIPTION
**Pull request by plaplant**
This PR adds a pre-commit hook for `black`. It also applies black-style formatting to the whole repo. This will make code style easy to maintain going forward.

Closes #5.

### CODE REVIEW

**Comment by plaplant** [comment]
@zacharymartinot there are a couple of issues that came up in the PR that I'd like to get your input on.
1. In `rime_funcs.py`, near the bottom, there is a mysterious import from an `interpolate` module (`interpolate.splprep` and `interpolate.splev`). Is this part of the Fortran compiled stuff? Or is it from scipy?
2. In `management.py`, there is an import of `__path__`, but I didn't see this defined in the `__int__.py` or `setup.py`. Is this defined somewhere?

I've also removed a few variable definitions that flake8 was reporting as not being used, but without tests running, it's not clear if these were important or breaking changes.

File: RIMEz/sky_models.py [hunk 1]
```diff
@@ -201,9 +200,9 @@ def alt_inner_blocks(I_s, RA_s, dec_s, L, ell_min, delta, Ilm_s):
 #
 #     return Ilm
 
-######### GLEAM
+# ######## GLEAM
```
Comment (steven-murray) [inline RIMEz/sky_models.py]: clearly this was black formatted... but I'd just get rid of the redundant #'s

File: RIMEz/sky_models.py [hunk 2]
```diff
@@ -201,9 +200,9 @@ def alt_inner_blocks(I_s, RA_s, dec_s, L, ell_min, delta, Ilm_s):
...
 
-######### diffuse sky model generation
+# ######## diffuse sky model generation
```
Comment (steven-murray) [inline RIMEz/sky_models.py]: reundant #'s

File: RIMEz/sky_models.py [hunk 3]
```diff
@@ -371,7 +370,7 @@ def linear_interp_rotation(hmap, R):
     return hp.get_interp_val(hmap, t, p)
 
 
-#### old thing
+# ### old thing
```
Comment (steven-murray) [inline RIMEz/sky_models.py]: #'s

**Comment by plaplant** [comment]
Thanks for reviewing @steven-murray! I've removed the repeated `#` characters.

**Comment by plaplant** [comment]
@zacharymartinot one other change I made on flake8's recommendation was to rename a variable in `management.py` from `I -> Iflux`. However, I noticed that this was being written to HDF5 files with a dataset named `I` as well, so I changed that to `Iflux` to be self-consistent. This does mean that we're not backwards compatible with files that have a dataset just named `I`, so we may want to build in some code to handle this if there are a lot of such files. Just wanted to let you know about this change.

**Pull request merged**